### DISCUSSION TITLE
Update the config file instead of overwriting

### DISCUSSION THREAD

**Discussion by resucutie** [ThioJoe/YT-Spammer-Purge#357] [Ideas]
Usually updating the YTSP config file is a pain in the ass due to it having to override. I had a suggestion of how we can avoid this issue
My idea is that you convert the current setting file into a JSON one to better file handling. And for updating you should get the current configs in the old config file and write a new one (but not overriding it) with the previous informations. So it would be in this way:

Imagine that we have those configs in the config file:
```json
{
    "setting": true
    "anotherSetting": 8
    "veryCoolSetting": "Magic"
}
```
And that you wanna add a config called `New_Stuff`. What you could do is to make a simple setting getter that creates a value in the file and returns a default value if it does not exist
```py
# Note that IDK how to code in python, but I have experience with JS
# Also this wasn't tested. Its just a example
import json
def getSetting(setting, defaultVal){
  if funcToCheckIfValueExists(setting):
    return funcToGetValue(setting)
  else:
    logicToAddSetting(setting, defaultVal)
    return defaultVal
}
def logicToAddSetting(setting, val){
  with open('settings.json') as outfile:
    oldSettings = json.load(outfile)
    oldSettings[setting] = val
    json.dump(oldSettings, outfile)
}

# When accessing a setting
veryCoolConfig = getSetting("Very_Cool_Configuration", "Magic")
```
This isn't perfect, since you lose your comments about the settings, but I guess this is worthy. For the comments, you could write in the wiki which setting does and which doesn't instead

**Comment by ThioJoe** [comment]
Yea that is definitely one that I intend to do soon. The current config is actually already parsable so it shouldn't need to be json, i just havent taken the time to implement it yet.

**Reply by resucutie** [reply]
Oh, I see. Thanks!

**Comment by ThioJoe** [comment]
The feature is now in the latest beta, 2.10.0-Beta1
